src/main/java/me/moodcat/backend/UserBackend.java

```diff
@@ -52,15 +50,25 @@ private MeModel retrieveMe(final String token) {
     }
 
     @Transactional
-    private User findOrRegisterUser(final MeModel me, final String token) {
+    private User findOrRegisterUser(final String token) {
         Preconditions.checkNotNull(token);
         
         final UserDAO userDAO = this.userDAOProvider.get();
+
+        try {
+            // Look for user in the caches
+            return userDAO.findByAccessToken(token);
+        }
+        catch (EntityNotFoundException e) {
+            // User not found in caches
```

Comment:
Maybe add a log statement indicating we are registering a new user for this token. An empty catch statement is not really nice.